iceoryx_hoofs/container/include/iox/detail/vector.inl

```diff
@@ -168,8 +190,15 @@ inline bool vector<T, Capacity>::emplace_back(Targs&&... args) noexcept
 {
     if (m_size < Capacity)
     {
-        // AXIVION Next Line AutosarC++19_03-A5.0.1, FaultDetection-IndirectAssignmentOverflow: Size guaranteed by T. Evaluation order is inconsequential.
-        new (&at(m_size++)) T(std::forward<Targs>(args)...);
+        if constexpr (std::is_trivial<T>::value)
+        {
+            at(m_size++) = T(std::forward<Targs>(args)...);
```

Comment:
```suggestion
            at(m_size++) = T{std::forward<Targs>(args)...};
```

This is necessary to make it work for this use case
```cpp
struct Point { uint32_t x; uint32_t y;};
iox::vector<Point, 10> points;
points.emplace_back(13, 42);
```

Reply:
done